DQMOffline/Trigger/plugins/HTMonitor.cc

```diff
@@ -0,0 +1,301 @@
+#include "DQMOffline/Trigger/plugins/HTMonitor.h"
+
+#include "CommonTools/TriggerUtils/interface/GenericTriggerEventFlag.h"
+
+#include "DataFormats/Math/interface/deltaPhi.h"
+
+// -----------------------------
+//  constructors and destructor
+// -----------------------------
+
+HTMonitor::HTMonitor( const edm::ParameterSet& iConfig ) : 
+  folderName_             ( iConfig.getParameter<std::string>("FolderName") )
+  , metToken_             ( consumes<reco::PFMETCollection>      (iConfig.getParameter<edm::InputTag>("met")       ) )   
+  , jetToken_             ( mayConsume<reco::PFJetCollection>      (iConfig.getParameter<edm::InputTag>("jets")      ) )   
+  , eleToken_             ( mayConsume<reco::GsfElectronCollection>(iConfig.getParameter<edm::InputTag>("electrons") ) )   
+  , muoToken_             ( mayConsume<reco::MuonCollection>       (iConfig.getParameter<edm::InputTag>("muons")     ) )   
+  , vtxToken_             ( mayConsume<reco::VertexCollection>      (iConfig.getParameter<edm::InputTag>("vertices")      ) )
+  , ht_variable_binning_ ( iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<std::vector<double> >("htBinning") )
+  , ht_binning_          ( getHistoPSet   (iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<edm::ParameterSet>   ("htPSet")    ) )
```

Comment:
@mtosi - we (me and @UBParker ) removed the fixed binning but then we realised that htvsLS 2D histogram depends on fixed binning, so we changed the definition of this histogram as - variable binning vs LS 2D histogram (https://github.com/UBParker/cmssw/blob/AddingHLTEl50PFJet165/DQMOffline/Trigger/plugins/METMonitor.cc#L117-#L126). I think the compilation error is coming because its not defined in the header (.h) file. After adding that in the .h file, I am trying to compile it locally but now I see that kind of format is not defined in the DQMServices/Core/interface/DQMStore.h file. So can we just keep the fixed binning definition for the htvsLS plot and also similarly for the metvsLS plot.